### PR TITLE
Add list of retire reasons

### PR DESCRIPTION
**Pull request by sircryptomaxi**
Requested in @Feedback. Links to issue [beefyfinance/beefy-app/issues/679](https://github.com/beefyfinance/beefy-app/issues/679).

To retire a vault with a reason, the strategist needs to add a key to the vault being retired: `retireReason`.

The current list would be:
`retireReason: 'rewards'` -> `RETIRED - Underlying platform stopped rewards.`
`retireReason: 'exploit'` -> `RETIRED - Token has been exploited, funds are SAFU in vault.`
`retireReason: 'tvl'` -> `RETIRED - TVL is too low.`

If that list is agreed, then translations would need to be made.

### CODE REVIEW

File: src/features/vault/components/PoolSummary/PoolSummary.js [hunk 1]
```diff
@@ -34,7 +34,13 @@ const PoolSummary = ({
...
     let state =
       pool.status === 'eol'
-        ? t('Vault-DepositsRetiredTitle')
+        ? pool.retireReason === 'rewards'
+          ? t('Vault-DepositsRetiredRewardsTitle')
+          : pool.retireReason === 'exploit'
+          ? t('Vault-DepositsRetiredExploitTitle')
+          : pool.retireReason === 'tvl'
+          ? t('Vault-DepositsRetiredTVLTitle')
+          : t('Vault-DepositsRetiredTitle')
```
Comment (Moonster-BSC) [inline src/features/vault/components/PoolSummary/PoolSummary.js:43]: Can you make this a map (object) for better readability. You would also be able to move the object to its own file.

I'm thinking replacing this with something like:

```ts
const retireReasonMap = {
  rewards: "Vault-DepositsRetiredRewardsTitle",
  exploit: "Vault-DepositsRetiredExploitTitle",
  // ...
}

const getRetireReason = (key) => {
  if (key in retireReasonMap) {
    const reason = retireReasonMap[key];
    return reason
  } else {
   return ""
  }
}
```

**Comment by Moonster-BSC** [comment]
Thanks for implementing this feedback item!! :)

File: src/features/vault/components/PoolSummary/PoolSummary.js (hunk 1 above)
Comment (sircryptomaxi) [reply src/features/vault/components/PoolSummary/PoolSummary.js]: I've implemented your suggestion, please have a look at the latest commit.

Comment (Moonster-BSC) [reply src/features/vault/components/PoolSummary/PoolSummary.js]: A maintainer will take a look at it and merge it if it looks good (I'm not a maintainer on this repo)

**Pull request merged**